src/Microsoft.Build.Utilities.ProjectCreation/PackageFileStream.cs

```diff
@@ -0,0 +1,100 @@
+﻿// Copyright (c) Jeff Kluge. All rights reserved.
+//
+// Licensed under the MIT license.
+
+#if !NETCOREAPP3_1
+
+using NuGet.Frameworks;
+
+#endif
+
+using NuGet.Packaging;
+using System;
+using System.IO;
+using System.Runtime.Versioning;
+
+namespace Microsoft.Build.Utilities.ProjectCreation
+{
+    /// <summary>
+    /// Represents a file to add to a NuGet package.
+    /// </summary>
+    internal class PackageFileStream : IPackageFile
+    {
+        private readonly Func<Stream> _streamFunc;
+        private string _path;
+
+        /// <summary>
+        /// Initializes a new instance of the <see cref="PackageFileStream" /> class.
+        /// </summary>
+        /// <param name="targetPath">The relative path to the file in the package.</param>
+        /// <param name="streamFunc">A <see cref="Func{TResult}" /> to be called to get a stream to the file.</param>
+        public PackageFileStream(string targetPath, Func<Stream> streamFunc)
```

Comment:
Curious as to why you choose `Func<Stream>` instead of simply receiving a `Stream` - is it just because it lend itself well within the abstraction together with the FileInfo.OpenRead? Or is there some performance / ownership thing i dont see?

Reply:
This is to make things "lazy" so that I don't new up a bunch of `Stream`s before the package is saved.  As you add files, only the delegate is added to the collection and when you call `Save`, the `Stream`s are opened one by one, copied, and closed.